src/foam/nanos/crunch/connection/CapabilityPayload.js

```diff
@@ -47,5 +48,44 @@ foam.CLASS({
         Validation errors for capability data objects.
       `
     }
-  ]
+  ],
+  methods: [
+      {
+        name: 'copyFrom',
+        type: 'CapabilityPayload',
+        args: [ { name: 'obj', type: 'FObject' } ],
```

Comment:
wonder is it the intention here to overload FObject.copyFrom?

Reply:
Yes, if we use the regular FObject copyFrom on a CapabilityPayload, the existing capabilityDataObjects map gets completely overwritten and we effectively lose the existing data we have.